src/components/ui/sort-by.tsx

```diff
@@ -1,17 +1,39 @@
-import { useState } from 'react';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
 
-import { AltArrowDown, SortVertical } from '../assets';
-import { cn } from '../lib/utils';
-import { Order, ProductField, SortOption } from '../types';
+import { AltArrowDown, SortVertical } from '../../assets';
+import { cn } from '../../lib/utils';
+import { Order, SortBy as SortByType, SortOption } from '../../types';
+import { SortState } from '../../app';
 
 interface SortByProps {
   options: SortOption[];
-  setSort: (sortBy: ProductField, order: Order) => void;
+  sort: SortState;
+  setSort: (sortBy: SortByType | undefined, order: Order | undefined) => void;
+  showFavorites: boolean;
+  setShowFavorites: Dispatch<SetStateAction<boolean>>;
 }
 
-export const SortBy = ({ options, setSort }: SortByProps) => {
+export const SortBy = ({
+  options,
+  sort,
+  setSort,
+  showFavorites,
+  setShowFavorites,
+}: SortByProps) => {
   const [open, setOpen] = useState(false);
-  const [label, setLabel] = useState('Sort By');
+  const [label, setLabel] = useState(
+    sort.sortBy && sort.order
+      ? options.find(
+          (opt) => opt.sortBy === sort.sortBy && opt.order === sort.order,
+        )?.label
+      : 'Sort By',
+  );
+
+  useEffect(() => {
+    if (!showFavorites && !sort.sortBy && !sort.order) {
+      setLabel('Sort By');
+    }
+  }, [showFavorites, sort]);
```

Comment:
I think this useEffect is not needed, the component will re-render if parent's state changes.

Reply:
The purpose of this useEffect is to execute when the showFavorites state changes in the App component. If I remove it, the functionality stops working. I haven’t been able to figure out a cleaner approach. Do you have any suggestions?